packages/catalog-client/src/CatalogClients.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, EntityName } from '@backstage/catalog-model';
+import {
+  CatalogClient,
+  EntitiesRequestOption,
+  EntitiesResponse,
+} from './types';
+import fetch from 'cross-fetch';
+import { Config } from '@backstage/config';
+import { EntityFilters } from './EntityFilters';
+
+/**
+ * A client to communicate with the catalog.
+ */
+export class CatalogClients implements CatalogClient {
+  /**
+   * Construct a catalog client given the app config.
+   *
+   * @param config The root of the app config
+   */
+  static fromConfig(config: Config): CatalogClient {
```

Comment:
Thinking this should be `fromDiscovery`, or simply just a `create` with a `discovery` option. The interface is the same for both [frontend](https://github.com/spotify/backstage/blob/21ba4d38a0f6359e8a08649b6c74bd098c457d23/packages/core-api/src/apis/definitions/DiscoveryApi.ts#L41) and [backend](https://github.com/spotify/backstage/blob/21ba4d38a0f6359e8a08649b6c74bd098c457d23/packages/backend-common/src/discovery/types.ts#L44), but best create a duplicate type here so we don't need to import either of those.

Reply:
I was gonna ask what to do about the discovery since it's not polyglot... Maybe we need a polyglot-common next to the backend-common and frontend-common (core)... ;)